src/managers/GuildMemberManager.js

```diff
@@ -135,6 +136,48 @@ class GuildMemberManager extends BaseManager {
     return this._fetchMany(options);
   }
 
+  /**
+   * Edits a member of the guild.
+   * <warn>The user must be a member of the guild</warn>
+   * @param {UserResolvable} user The member to edit
+   * @param {GuildMemberEditData} data The data to edit the member with
+   * @param {string} [reason] Reason for editing this user
+   * @returns {Promise<void>}
+   */
+  async edit(user, data, reason) {
+    const id = this.client.users.resolveID(user);
+    if (!id) throw new TypeError('INVALID_TYPE', 'user', 'UserResolvable');
+
+    if (data.channel) {
+      data.channel = this.guild.channels.resolve(data.channel);
+      if (!data.channel || data.channel.type !== 'voice') {
+        throw new Error('GUILD_VOICE_CHANNEL_RESOLVE');
+      }
+      data.channel_id = data.channel.id;
+      data.channel = undefined;
+    } else if (data.channel === null) {
+      data.channel_id = null;
+      data.channel = undefined;
+    }
+    if (data.roles) data.roles = data.roles.map(role => (role instanceof Role ? role.id : role));
+    let endpoint = this.client.api.guilds(this.guild.id);
+    if (id === this.client.user.id) {
+      const keys = Object.keys(data);
+      if (keys.length === 1 && keys[0] === 'nick') endpoint = endpoint.members('@me').nick;
+      else endpoint = endpoint.members(id);
+    } else {
+      endpoint = endpoint.members(id);
+    }
+    await endpoint.patch({ data, reason }).then(d => {
+      if (this.cache.has(id)) {
```

Comment:
Shouldn't we cache with the new data we get from the member? Assuming, of course, if we get any data..

Reply:
Valid point - I'll check if this would be possible. The idea was that when operating on an uncached instance it would remain uncached, but if patch returns a complete member object to us I agree we should add to cache. Will test and implement if so.

Reply:
> PATCH/guilds/{guild.id}/members/{user.id}
Modify attributes of a guild member. Returns a 204 empty response on success.

We do not, so no change here.